cli/src/common/boto3/ec2.py

```diff
@@ -98,6 +98,15 @@ def image_exists(self, image_name):
                 return False
             raise e
 
+    def describe_image_by_tag(self, image_name):
+        """Return a dict of image info by searching image name tag as filter."""
+        filters = [{"Name": "tag:" + PCLUSTER_IMAGE_NAME_TAG, "Values": [image_name]}]
+        owners = ["self"]
+        try:
+            return self.describe_images(ami_ids=[], filters=filters, owners=owners)[0]
+        except AWSClientError:
+            return None
```

Comment:
are we ok to handle all the errors the same way: e.g. unexisting ami, missing IAM permissions, throttling?

Reply:
I think there is a similar issue with the image_exists method above

Reply:
Updated, the idea here is we always throw an AWSClientError for describe_image or describe_stack function if boto3 call fails, and catch this AWSClientError in logic function which calls these boto3 methods, and check the error message. If the error message contains `stack id doesn't exist` or `No image matching the search criteria found`, which means it is the case of unexisting ami/stack. Otherwise, we raise the error for other errors like missing IAM permissions, throttling.

In image_exists and stack_exists functions, we only return False if the case is unexisting ami/stack.